ably.ts.d

```diff
@@ -0,0 +1,400 @@
+  message: String;
+  statusCode: number;
+}
+
+interface StatsMessageTypes {
+  all: number,
+  messages: number,
+  presence: number;
+}
+
+interface StatsRequestCount {
+  failed: number,
+  refused: number,
+  succeeded: number
+}
+
+interface StatsResourceCount {
+  mean: number,
+  min: number,
+  opened: number,
+  peak: number,
+  refused: number
+}
+
+interface StatsConnectionTypes {
+  all: number,
+  plain: number,
+  tls: number
+}
+
+interface StatsMessageTraffic {
+  all: number,
+  realtime: number,
+  rest: number,
+  webhook: number
+}
+
+interface TokenDetails {
+  capability: String;
+  clientId: String;
+  expires: number; // Time
+  issued: number; // Time
+  token: String;
+}
+
+interface TokenRequest {
+  capability: String;
+  clientId?: String;
+  keyName: String;
+  mac: String;
+  nonce: String;
+  timestamp: number; // Time
+  ttl?: number;
+}
+
+interface ChannelOptions {
+  cipher: any;
+}
+
+interface RestPresenceHistoryParams {
+  start?: number;
+  end?: number;
+  direction?: String;
+  limit?: number;
+}
+
+
+interface RestPresenceParams {
+  limit?: number;
+  clientId?: String;
+  connectionId?: String;
+}
+
+interface RealtimePresenceParams {
+  waitForSync?: boolean;
+  clientId?: String;
+  connectionId?: String;
+}
+
+interface RealtimePresenceHistoryParams {
+  start?: number;
+  end?: number;
+  direction?: String;
+  limit?: number;
+  untilAttach?: boolean
+}
+
+interface LogInfo {
+  level?: number;
+  handler?: (...args) => void;
+}
+
+
+// Common Listeners
+type ListenerFn = (...args: Array<any>) => void;
+type PaginatedResultCallback = (Error: ErrorInfo, results: PaginatedResult ) => void;
+type standardCallback = (Error: ErrorInfo, results: any) => void;
+type messageCallback = (Msg: any) => void;
+type errorCallback = (Error: ErrorInfo) => void;
+
+
+
+// Internal Classes
+declare class EventEmitter {
+  on: (eventOrCallback: String | ListenerFn, callback?: ListenerFn) => void;
+
+  once: (eventOrCallback: String | ListenerFn, callback?: ListenerFn) => void;
+
+  off: (eventOrCallback?: String | ListenerFn, callback?: ListenerFn) => void;
+}
+
+// Classes
+export declare class Auth {
+  clientId: String;
+  authorize: (tokenParams?: TokenParams, authOptions?: AuthOptions, callback?: (Error: ErrorInfo, Results: TokenDetails) => void) => void;
+  createTokenRequest: (tokenParams?: TokenParams, authOptions?: AuthOptions, callback?: (Error: ErrorInfo, Results: TokenRequest) => void) => void;
+  requestToken: (TokenParams?: TokenParams, authOptions?: AuthOptions, callback?: (Error: ErrorInfo, Results: TokenDetails) => void) => void;
+}
+
+export declare class Presence {
+  get: (params: RestPresenceParams, PaginatedResultCallback) => void;
+
+  history: (params: RestPresenceHistoryParams, PaginatedResultCallback) => void;
+}
+
+export declare class RealtimePresence {
+  syncComplete: () =>  boolean;
+
+  get: (Params: RealtimePresenceParams, callback?: PaginatedResultCallback) => void;
```

Comment:
@NathanaelA the reason it doesn't is because the Realtime client keeps an up to date set of all presence members in memory, whereas the REST version obtains it from the server upon request.  So in the case of Realtime there is no need for paging given the set is in memory anyway